routers/web/auth/auth.go

```diff
@@ -424,6 +424,8 @@ func SignUp(ctx *context.Context) {
 
 	ctx.Data["SignUpLink"] = setting.AppSubURL + "/user/sign_up"
 
+	ctx.Data["IsFirstTimeRegistration"] = user_model.CountUsers(ctx, nil) == 0
```

Comment:
If the instance has large amount of users, the query is a full-table scan and quite slow.